examples/TTI_codegen.py

```diff
@@ -62,12 +63,25 @@ def damp_boundary(damp):
                               dtype=self.dtype, nbpml=nbpml)
         self.src.data[:] = data.get_source()[:, np.newaxis]
 
-    def Forward(self, save=False, cache_blocking=None):
+    def Forward(self, save=False, cse=True, auto_tuning=False,
+                cache_blocking=None, compiler=None):
         fw = ForwardOperator(self.model, self.src, self.damp, self.data,
                              time_order=self.t_order, spc_order=self.s_order,
-                             save=save, cache_blocking=cache_blocking)
+                             profile=True, save=save, cache_blocking=cache_blocking,
+                             cse=cse, compiler=compiler)
+
+        if auto_tuning:
+            fw_new = ForwardOperator(self.model, self.src, self.damp, self.data,
+                                     time_order=self.t_order, spc_order=self.s_order,
+                                     profile=True, save=save, cse=cse, compiler=compiler)
+
+            at = AutoTuner(fw_new)
+            at.auto_tune_blocks(self.s_order + 1, self.s_order * 4 + 2)
+            fw.propagator.cache_blocking = at.block_size
+
         u, v, rec = fw.apply()
-        return (rec.data, u.data, v.data)
+        return (rec.data, u.data, v.data,
+                fw.propagator.gflops, fw.propagator.oi)
 
```

Comment:
More of a personal view, but I don't think`gflops` and `oi` should be default output.  The forward operator is there for modelling purpose, and extra output just make its use messy. Just a thought doesn't need to be changed now, but I will for Users at some point.